### PR TITLE
Preserve whitespace in elements containing text

### PR DESCRIPTION
**Pull request by chromakode**
### Problem

SVGO configures its SAX parser to normalize whitespace, [collapsing multiple space characters into one](https://github.com/isaacs/sax-js/blob/5aee2163d55cff24b817bbf550bac44841f9df45/lib/sax.js#L640). 

This can change the semantic meaning of SVG content:

- In `<text>` and `<tspan>` nodes when whitespace is being preserved (e.g. `xml:space="preserve"` or `style="white-space: pre"`).
- In `<text>` and `<tspan>` nodes where [glyphs are spaced explicitly using multiple `x` values](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/x#tspan). Changing the number of characters in the node changes the indexes of characters (see https://github.com/svg/svgo/issues/1044).

### Solution

This patch disables space normalization in `sax`. It removes trimming for node names classified as "textContent" in `_collections.js`. In addition, it changes the pretty printing logic to not add indents or whitespace when outputting such nodes.

I needed to update a few text fixtures which captured the previous space normalizing behavior. Existing plugin tests have a mixture of text nodes containing and not containing extra whitespace, so they cover the change in functionality. I've also added a test to `svg2js` to verify that the whitespace is preserved when parsing SVGs.

### Related issues
- https://github.com/svg/svgo/issues/634
- https://github.com/svg/svgo/issues/1044
- https://github.com/svg/svgo/issues/1067

### CODE REVIEW

File: test/plugins/inlineStyles.13.svg [hunk 1]
```diff
@@ -12,9 +12,7 @@
             /* Atrules with block */
             @font-face {
                 font-family: SomeFont;
-                src: local("Some Font"),
-                local("SomeFont"),
-                url(SomeFont.ttf);
+                src: local("Some Font"), local("SomeFont"), url(SomeFont.ttf);
```
Comment (chromakode) [inline test/plugins/inlineStyles.13.svg]: `csso` preserves the newlines in this `src` value, which were previously being turned into spaces by the `sax` normalization.

**Comment by felixfbecker** [comment]
@GreLI It would be really great to see this merged 🙇

**Comment by TrySound** [comment]
Please rebase

**Comment by chromakode** [comment]
@TrySound rebased :+1:

File: lib/svgo/svg2js.js [hunk 2]
```diff
@@ -115,8 +116,8 @@ module.exports = function(data) {
         elem = pushToContent(elem);
         current = elem;
 
-        // Save info about <text> tag to prevent trimming of meaningful whitespace
-        if (data.name == 'text' && !data.prefix) {
+        // Save info about tags containing text to prevent trimming of meaningful whitespace
+        if (textElems.indexOf(data.name) !== -1 && !data.prefix) {
```
Comment (TrySound) [inline lib/svgo/svg2js.js:120]: Use includes array method please

File: lib/svgo/svg2js.js [hunk 3]
```diff
@@ -4,12 +4,13 @@ var SAX = require('sax'),
     JSAPI = require('./jsAPI.js'),
     CSSClassList = require('./css-class-list'),
     CSSStyleDeclaration = require('./css-style-declaration'),
-    entityDeclaration = /<!ENTITY\s+(\S+)\s+(?:'([^']+)'|"([^"]+)")\s*>/g;
+    entityDeclaration = /<!ENTITY\s+(\S+)\s+(?:'([^']+)'|"([^"]+)")\s*>/g,
+    textElems = require('../../plugins/_collections.js').textElems;
```
Comment (TrySound) [inline lib/svgo/svg2js.js:8]: Put require before regexp

**Comment by TrySound** [comment]
Thanks!
